index.d.ts

```diff
@@ -115,16 +118,30 @@ declare namespace Cmd {
     args?: any[]
   ): MapCmd<A>;
 
-  export function run<A extends Action, B extends Action>(
-    f: Function,
+  type InferArgs<T> = T extends (...args: infer T) => any ? T : never;
+  // Allow the use of special dispatch | getState symbols
+  type ArgOrSymbol<T> = {
+    [K in keyof T]: T[K] extends Dispatch | GetState
+      ? typeof dispatch | typeof getState | T[K]
+      : T[K];
+  }
```

Comment:
To be honest I didn't think that the type system could express `Cmd.run` API was fully. I'm glad to learn that TypeScript is this powerful, but if we have to resort to this much magic in the type-definitions, maybe we should revisit if this API can be simplified first.